src/main/java/uk/gov/hmcts/darts/audio/service/impl/AudioTransformationServiceImpl.java

```diff
@@ -51,4 +55,14 @@ public TransientObjectDirectoryEntity saveTransientDataLocation(MediaRequestEnti
         return transientObjectDirectoryService.saveTransientDataLocation(mediaRequest, externalLocation);
     }
 
+    @Override
+    public Optional<UUID> getMediaLocation(MediaEntity media) {
+        Optional<UUID> externalLocation = Optional.empty();
+        List<ExternalObjectDirectoryEntity> externalObjectDirectoryEntityList = media.getExternalObjectDirectoryEntityList();
+        if (!externalObjectDirectoryEntityList.isEmpty()) {
+            externalLocation = Optional.ofNullable(externalObjectDirectoryEntityList.get(0).getExternalLocation());
```

Comment:
Is it possible there could be `> 1` items in `externalObjectDirectoryEntityList`?

If there's an assumption there will only ever be `<= 1` items, it might be worth adding a size check and logging a warning if `size > 1`.

Reply:
@hemantasharma1129 confirmed that we should also be checking some other columns on the `external_object_directory` table, so I have added a custom finder `ExternalObjectDirectoryRepository#findByMediaStatusAndType` to do that in commit bb4c5e0